Makefile

```diff
@@ -129,7 +129,7 @@ poetry-lock: $(POETRY)
 # Update the kserve package first as other packages depends on it.
 	cd ./python && \
 	cd kserve && $(POETRY) lock --no-update && cd .. && \
-	for file in $$(find . -type f -name "pyproject.toml" -not -path "./pyproject.toml"); do \
+	for file in $$(find . -type f -name "pyproject.toml" -not -path "./pyproject.toml" -not -path "*/.venv/*"); do \
```

Comment:
I've faced this same issue, and I kinda fixed it on another PR that I am going to send soon.